### PR TITLE
Add temp directory to W&B tests

### PR DESCRIPTION
**Pull request by skanjila**
# Code Pull Requests

Please provide the following:

- This PR establishes a temp directory for wandb

# Documentation Pull Requests

N/A

### CODE REVIEW

**Comment by skanjila** [comment]
Output of python -m pytest tests/integration_tests/test_contrib_wandb.py::test_contrib_experiment -vs

/Users/saikatkanjilalmbpro/code/adobe/rtml-python-runtime/venv/bin/python /Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/_jb_pytest_runner.py --target test_contrib_wandb.py::test_contrib_experiment
Testing started at 2:51 PM ...
Launching pytest with arguments test_contrib_wandb.py::test_contrib_experiment --no-header --no-summary -q in /Users/saikatkanjilalmbpro/code/opensource/ludwig/tests/integration_tests

============================= test session starts ==============================
collecting ... collected 1 item

test_contrib_wandb.py::test_contrib_experiment 

======================== 1 passed, 1 warning in 12.56s =========================

Process finished with exit code 0
PASSED                    [100%]Experiment name: experiment
Model name: run
Output directory: results/experiment_run_0


ludwig_version: '0.4'
command: ('/Users/saikatkanjilalmbpro/code/opensource/ludwig/tests/integration_tests/scripts/run_train_wandb.py '
 '--csv-filename '
 '/var/folders/16/jmt05v316g55sv5d1hpt6rgm0000gr/T/tmptrji8qwd/C1482409D0.csv')
commit_hash: 'b66842ba429c'
random_seed: 42
dataset: '/var/folders/16/jmt05v316g55sv5d1hpt6rgm0000gr/T/tmptrji8qwd/C1482409D0.csv'
data_format: 'csv'
config: {   'combiner': {'fc_size': 14, 'type': 'concat'},
    'input_features': [   {   'column': 'image_4BD43',
                              'destination_folder': '/Users/saikatkanjilalmbpro/code/opensource/ludwig/tests/integration_tests/generated_images',
                              'encoder': 'resnet',
                              'fc_size': 8,
                              'name': 'image_4BD43',
                              'num_filters': 8,
                              'preprocessing': {   'height': 12,
                                                   'in_memory': True,
                                                   'num_channels': 3,
                                                   'width': 12},
                              'proc_column': 'image_4BD43_PMr3_m',
                              'resnet_size': 8,
                              'tied': None,
                              'type': 'image'}],
    'output_features': [   {   'column': 'category_5EEF9',
                               'dependencies': [],
                               'embedding_size': 5,
                               'idx2str': [   'AoJ',
                                              'HXmZkD',
                                              'Kn',
                                              'jGEjMQ',
                                              'wKTBFldnuK',
                                              'SslJQsFkLS',
                                              'swfUdn',
                                              'vjDMTEO',
                                              'YgQ',
                                              'WnSOEH'],
                               'loss': {   'class_similarities_temperature': 0,
                                           'class_weights': 1,
                                           'confidence_penalty': 0,
                                           'labels_smoothing': 0,
                                           'robust_lambda': 0,
                                           'type': 'softmax_cross_entropy',
                                           'weight': 1},
                               'name': 'category_5EEF9',
                               'proc_column': 'category_5EEF9_mZFLky',
                               'reduce_dependencies': 'sum',
                               'reduce_input': 'sum',
                               'top_k': 3,
                               'type': 'category',
                               'vocab_size': 10}],
    'preprocessing': {   'audio': {   'audio_feature': {'type': 'raw'},
                                      'audio_file_length_limit_in_s': 7.5,
                                      'in_memory': True,
                                      'missing_value_strategy': 'backfill',
                                      'norm': None,
                                      'padding_value': 0},
                         'bag': {   'fill_value': '<UNK>',
                                    'lowercase': False,
                                    'missing_value_strategy': 'fill_with_const',
                                    'most_common': 10000,
                                    'tokenizer': 'space'},
                         'binary': {   'fill_value': 0,
                                       'missing_value_strategy': 'fill_with_const'},
                         'category': {   'fill_value': '<UNK>',
                                         'lowercase': False,
                                         'missing_value_strategy': 'fill_with_const',
                                         'most_common': 10000},
                         'date': {   'datetime_format': None,
                                     'fill_value': '',
                                     'missing_value_strategy': 'fill_with_const'},
                         'force_split': False,
                         'h3': {   'fill_value': 576495936675512319,
                                   'missing_value_strategy': 'fill_with_const'},
                         'image': {   'in_memory': True,
                                      'infer_image_dimensions': True,
                                      'infer_image_max_height': 256,
                                      'infer_image_max_width': 256,
                                      'infer_image_num_channels': True,
                                      'infer_image_sample_size': 100,
                                      'missing_value_strategy': 'backfill',
                                      'num_processes': 1,
                                      'resize_method': 'interpolate',
                                      'scaling': 'pixel_normalization'},
                         'numerical': {   'fill_value': 0,
                                          'missing_value_strategy': 'fill_with_const',
                                          'normalization': None},
                         'sequence': {   'fill_value': '<UNK>',
                                         'lowercase': False,
                                         'missing_value_strategy': 'fill_with_const',
                                         'most_common': 20000,
                                         'padding': 'right',
                                         'padding_symbol': '<PAD>',
                                         'sequence_length_limit': 256,
                                         'tokenizer': 'space',
                                         'unknown_symbol': '<UNK>',
                                         'vocab_file': None},
                         'set': {   'fill_value': '<UNK>',
                                    'lowercase': False,
                                    'missing_value_strategy': 'fill_with_const',
                                    'most_common': 10000,
                                    'tokenizer': 'space'},
                         'split_probabilities': (0.7, 0.1, 0.2),
                         'stratify': None,
                         'text': {   'char_most_common': 70,
                                     'char_sequence_length_limit': 1024,
                                     'char_tokenizer': 'characters',
                                     'char_vocab_file': None,
                                     'fill_value': '<UNK>',
                                     'lowercase': True,
                                     'missing_value_strategy': 'fill_with_const',
                                     'padding': 'right',
                                     'padding_symbol': '<PAD>',
                                     'pretrained_model_name_or_path': None,
                                     'unknown_symbol': '<UNK>',
                                     'word_most_common': 20000,
                                     'word_sequence_length_limit': 256,
                                     'word_tokenizer': 'space_punct',
                                     'word_vocab_file': None},
                         'timeseries': {   'fill_value': '',
                                           'missing_value_strategy': 'fill_with_const',
                                           'padding': 'right',
                                           'padding_value': 0,
                                           'timeseries_length_limit': 256,
                                           'tokenizer': 'space'},
                         'vector': {   'fill_value': '',
                                       'missing_value_strategy': 'fill_with_const'}},
    'training': {   'batch_size': 128,
                    'bucketing_field': None,
                    'decay': False,
                    'decay_rate': 0.96,
                    'decay_steps': 10000,
                    'early_stop': 5,
                    'epochs': 2,
                    'eval_batch_size': None,
                    'gradient_clipping': None,
                    'increase_batch_size_on_plateau': 0,
                    'increase_batch_size_on_plateau_max': 512,
                    'increase_batch_size_on_plateau_patience': 5,
                    'increase_batch_size_on_plateau_rate': 2,
                    'learning_rate': 0.001,
                    'learning_rate_warmup_epochs': 1,
                    'optimizer': {   'betas': (0.9, 0.999),
                                     'eps': 1e-08,
                                     'type': 'adam'},
                    'reduce_learning_rate_on_plateau': 0,
                    'reduce_learning_rate_on_plateau_patience': 5,
                    'reduce_learning_rate_on_plateau_rate': 0.5,
                    'regularization_lambda': 0,
                    'regularization_type': 'l2',
                    'staircase': False,
                    'validation_field': 'combined',
                    'validation_metric': 'loss'}}
torch_version: '1.10.0'


Using full raw dataset, no hdf5 and json file with the same name have been found
Building dataset (it may take a while)
Training set: 18
Validation set: 2
Test set: 4
wandb.on_train_init() called...
wandb.on_train_start() called...

╒══════════╕
│ TRAINING │
╘══════════╛


Epoch 1
Training: 100% 1/1 [00:00<00:00, 10.96it/s]
Evaluation train: 100% 1/1 [00:00<00:00, 47.67it/s]
Evaluation vali : 100% 1/1 [00:00<00:00, 130.68it/s]
Evaluation test : 100% 1/1 [00:00<00:00, 129.02it/s]
Took 0.1388s
╒══════════════════╤════════╤════════════╤═════════════╕
│ category_5EEF9   │   loss │   accuracy │   hits_at_k │
╞══════════════════╪════════╪════════════╪═════════════╡
│ train            │ 1.9182 │     0.2222 │      0.5000 │
├──────────────────┼────────┼────────────┼─────────────┤
│ vali             │ 1.9958 │     0.5000 │      0.5000 │
├──────────────────┼────────┼────────────┼─────────────┤
│ test             │ 2.2810 │     0.0000 │      0.0000 │
╘══════════════════╧════════╧════════════╧═════════════╛
╒════════════╤════════╕
│ combined   │   loss │
╞════════════╪════════╡
│ train      │ 1.9182 │
├────────────┼────────┤
│ vali       │ 2.0735 │
├────────────┼────────┤
│ test       │ 2.8512 │
╘════════════╧════════╛


Epoch 2
Training: 100% 1/1 [00:00<00:00, 45.26it/s]
Evaluation train: 100% 1/1 [00:00<00:00, 92.03it/s]
Evaluation vali : 100% 1/1 [00:00<00:00, 117.68it/s]
Evaluation test : 100% 1/1 [00:00<00:00, 129.77it/s]
Took 0.5432s
╒══════════════════╤════════╤════════════╤═════════════╕
│ category_5EEF9   │   loss │   accuracy │   hits_at_k │
╞══════════════════╪════════╪════════════╪═════════════╡
│ train            │ 2.1492 │     0.3333 │      0.7778 │
├──────────────────┼────────┼────────────┼─────────────┤
│ vali             │ 2.1220 │     0.5000 │      0.5000 │
├──────────────────┼────────┼────────────┼─────────────┤
│ test             │ 2.2326 │     0.0000 │      0.0000 │
╘══════════════════╧════════╧════════════╧═════════════╛
╒════════════╤════════╕
│ combined   │   loss │
╞════════════╪════════╡
│ train      │ 1.7540 │
├────────────┼────────┤
│ vali       │ 2.0131 │
├────────────┼────────┤
│ test       │ 2.7859 │
╘════════════╧════════╛

Best validation model epoch: 2
Best validation model loss on validation set combined: 2.013113498687744
Best validation model loss on test set combined: 2.785886764526367

Finished: experiment_run
Saved to: results/experiment_run_0
Evaluation: 100% 1/1 [00:00<00:00, 159.16it/s]

===== category_5EEF9 =====
accuracy: 0.0
hits_at_k: 0.0
loss: 2.311676025390625

File: tests/integration_tests/test_contrib_wandb.py [hunk 1]
```diff
@@ -14,7 +14,13 @@
 
 
 @pytest.mark.distributed
-def test_contrib_experiment(csv_filename):
+def test_contrib_experiment(csv_filename, tmpdir):
+    wandb_dir = os.path.join(tmpdir, 'results')
+    is_exist = os.path.exists(wandb_dir)
```
Comment (tgaddair) [inline tests/integration_tests/test_contrib_wandb.py:19]: Nit: can simplify these 3 lines to:

```
os.makedirs(wandb_dir, exist_ok=True)
```

Comment (skanjila) [reply tests/integration_tests/test_contrib_wandb.py]: simplified

**Review by tgaddair — APPROVED** [review]
